### PR TITLE
Added mwlib services

### PR DESCRIPTION
**Pull request by ip1981**
Simple local usage:

```
mwlib = {
  nserve = {
    enable = true;
  };

  qserve = {
    enable = true;
  };
};
```

For MediaWiki:

```
  $wgCollectionMWServeURL = 'http://localhost:8899';
  $wgCollectionFormats = array(
    'rl' => 'PDF',
    'odf' => 'ODT',
  );

  $wgCollectionPortletFormats = array (
    'odf', 'rl'
  );
```

### CODE REVIEW

File: nixos/modules/services/misc/mwlib.nix [hunk 1]
```diff
@@ -0,0 +1,256 @@
+{ config, lib, pkgs, ... }:
+let
+  pypkgs = pkgs.python27Packages;
+
+  inherit (lib) types mkOption mkIf optionalString
```
Comment (lethalman) [inline nixos/modules/services/misc/mwlib.nix]: Just do `with lib;`

File: nixos/modules/services/misc/mwlib.nix [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                concatStringsSep concatMapStringsSep;
+  inherit (config.services.mwlib) nserve qserve nslave;
```
Comment (lethalman) [inline nixos/modules/services/misc/mwlib.nix]: Do `cfg = config.services.mwlib;` and use that.

File: nixos/modules/services/misc/mwlib.nix (hunk 1 above)
Comment (ip1981) [inline nixos/modules/services/misc/mwlib.nix]: here we go

**Comment by lethalman** [comment]
Ok haven't tested it but looks fine, thanks.

**Comment by ip1981** [comment]
You can imaging how hard I did test it :-)

**Pull request merged**